### ISSUE TITLE
Apply on the module changes between Linux and MacOS

### ISSUE THREAD

**Issue by spoukke** [terraform-google-modules/terraform-google-event-function#105]
### TL;DR

The `google_storage_bucket_object.main` must be replaced if we apply first on Linux then on Mac (or conversly). This comes from the `md5` of the `archive_file` data being different between the two operating systems because the generated `zip` file does not have the same system permission, `0664` on linux and `0644` on macOS.

### Expected behavior

The resource  `google_storage_bucket_object.main` should not diverge between two operating systems. The [output_file_mode](https://registry.terraform.io/providers/hashicorp/archive/latest/docs/data-sources/archive_file#output_file_mode) field on the data `archive_file` should be a defaulted variable to avoid this issue.

### Terraform Configuration

```hcl
module "function" {
  source  = "terraform-google-modules/event-function/google"
  version = "~> 2.3.0"

  entry_point = "ProcessPubSub"
  event_trigger = {
    event_type = "google.pubsub.topic.publish"
    resource   = module.pubsub_topic.id
  }
  name             = "${local.name}-exporter"
  project_id       = var.project
  region           = var.region
  runtime          = "go116"
  source_directory = "${path.module}/function"

  source_dependent_files = []

  available_memory_mb                = 256
  bucket_force_destroy               = true
  bucket_name                        = "${local.name}-scheduled-exporter-function-${random_id.suffix.hex}"
  description                        = "Function to backup SQL instance."
  event_trigger_failure_policy_retry = false
  service_account_email              = google_service_account.this.email
  timeout_s                          = 540
}
```
```


### Terraform Version

```sh
1.3.0
```
